ibllib/io/video.py

```diff
@@ -63,20 +64,18 @@ def get_video_frames_preload(video_path, frame_numbers=None, mask=Ellipsis, as_l
         frame_images = [None] * len(frame_numbers)
     else:
         ret, frame = cap.read()
-        frame_images = np.empty((len(frame_numbers), *func(frame[mask or ...]).shape),
-                                np.uint8)
+        frame_images = np.zeros((len(frame_numbers), *func(frame[mask or ...]).shape), np.uint8)
+    print(f"Loading {len(frame_numbers)} frames")
     for ii, i in enumerate(frame_numbers):
-        sys.stdout.write(f'\rloading frame {ii}/{len(frame_numbers)}')
```

Comment:
Please restore the frame loader.  What were @oliche's reasons for removing it?  I'm fine with a quiet flag that is False by default.

Reply:
The argument was that it slows things down to print to the command line

Reply:
Printing to the stdout is _extremely_ fast.  In fact it's even faster than a print statement:
```
import time
import sys
n = 10000
times = []
print('========== Starting print test ==========')
for _ in range(3):
    start_time = time.time()
    for i in range(n):
        print(f'\rloading frame {i}/{n}')
    times.append(time.time() - start_time)
print(times)
print(sum(times) / 3)
times = []
print('========== Starting stdout test ==========')
for _ in range(3):
    start_time = time.time()
    for i in range(n):
        sys.stdout.write(f'\rloading frame {i}/{n}')
        sys.stdout.flush()
    times.append(time.time() - start_time)
print(times)
print(sum(times) / 3)
"""
========== Starting print test ==========
[5.338913440704346, 1.0393381118774414, 1.0872581005096436]
2.4885032176971436
========== Starting stdout test ==========
loading frame 9999/10000[0.47832822799682617, 0.4129786491394043, 0.4161257743835449]
0.4358108838399251"""
```

Also it's extremely useful when you're sampling non-contiguous frames, especially over http.  That can take many minutes so it's good to know where you are.  The camera QC samples 100 frames by default and that takes over 10 minutes.

Reply:
I'm pretty confident that any performance improvement you see is a result of not having to re-open the video file.  I don't know how many times you call this function but that quickly adds up, as does getting the video length.  (FYI the VideoStreamer object stores it as an attribute).  When I wrote the function I intended for it to be called once per video so I didn't optimize it for other use cases.

Reply:
Should all be solved with the quiet flag, so one can pick their preference